### DISCUSSION TITLE
Feature request/bugfix

### DISCUSSION THREAD

**Discussion by gillett-hernandez** [gillett-hernandez/Foreman2#4] [General]
Add recipe main product to preset so that icons are set appropriately. Requires changes to the export mod, and changes to the preset loader system.

**Comment by topennanen** [comment]
Here's a diff to fix this as well as the mining fluid amounts (currently 10 times what they are in game) , goes in all of the "Mods\foremanexport_1.0.0\instrument-control - XX.lua" files, where XX are nn, ne, en, ee for expensive research and recipe settings
```
79c79,81
<               if recipe.products[1] then
---
>               if recipe.main_product then
>                       trecipe["icon_alt_name"] = 'icon.i.' .. recipe.main_product.name
>               elseif recipe.products[1] then
443c445
<                               tresource['fluid_amount'] = resource.mineable_properties.fluid_amount
---
>                               tresource['fluid_amount'] = resource.mineable_properties.fluid_amount / 10.0
```

**Comment by gillett-hernandez** [comment]
thanks, i'm testing it now and will likely publish a new release with this patch soon.
